programs/drift/src/math/margin.rs

```diff
@@ -339,21 +328,43 @@ pub fn calculate_margin_requirement_and_total_collateral_and_liability_info(
             }
         } else {
             let signed_token_amount = spot_position.get_signed_token_amount(&spot_market)?;
-            let (worst_case_token_amount, worst_case_orders_value): (i128, i128) = spot_position
-                .get_worst_case_token_amount(
-                    &spot_market,
-                    oracle_price_data,
-                    if strict {
-                        Some(
-                            spot_market
-                                .historical_oracle_data
-                                .last_oracle_price_twap_5min,
-                        )
-                    } else {
-                        None
-                    },
-                    Some(signed_token_amount),
-                )?;
+
+            let OrderFillSimulation {
+                token_amount: worst_case_token_amount,
+                orders_value: mut worst_case_orders_value,
+                token_value: worst_case_token_value,
+                weighted_token_value: mut worst_case_weighted_token_value,
+                ..
+            } = spot_position.get_worst_case_token_amount(
+                &spot_market,
+                &strict_oracle_price,
+                Some(signed_token_amount),
+                margin_requirement_type,
+            )?;
+
+            // Need to recalculate weighted_token_value if user_custom_margin_ratio != 0
+            if user_custom_margin_ratio != 0 {
```

Comment:
should this live in the `get_worst_case_token_amount` function?

Reply:
was thinking we don't want the user margin ratio to effect other conditional logic that depends on get_worst_case_token_amount, wdyt?